tests/test_listener.py

```diff
@@ -0,0 +1,72 @@
+# Source pytest module
```

Comment:
Typo here

Reply:
Wait, we have Source pytest module for `test_context` also LOL

Reply:
Probably missed that, fix it then.

Reply:
Could you please add more test and also fix the typo in `test_context`?